static/app/types/hooks.tsx

```diff
@@ -658,6 +659,18 @@ type SidebarNavigationItemHook = () => React.ComponentType<{
   id: string;
 }>;
 
+/**
+ * Insights upsell hook takes in a insights sidebar item id
+ * and (if applicable) passes in the module specific upsell page as props to its children.
+ */
+type InsightsUpsellHook = () => React.ComponentType<{
+  children: (opts: {
+    disabled: boolean;
+    upsellPage: React.ReactElement | null;
+  }) => React.ReactElement;
+  id: string;
```

Comment:
Out of curiosity, does this need to use the render props approach? Are there props that have to be passed down, or is it possible to use a "regular" hook that just substitutes in other content?

Reply:
I'll verify this more tmrw. But I believe the regular disabled feature hooks don't allow you to pass anything into them. In our case we want to pass in the module name to show the appropriate upsell page.

Reply:
Ok i dug around, its possible with a "component hook", made the changes.